### PR TITLE
Avoid git rebase conflicts during development on other branches

### PR DESCRIPTION
**Pull request by nagmat84**
As discussed on Gitter.

### CODE REVIEW

**Comment by nagmat84** [comment]
I don't understand why `codecov/patch` is unhappy.

**Review by ildyria — CHANGES_REQUESTED** [review]
https://github.com/LycheeOrg/Lychee/blob/master/app/Metadata/GitHubFunctions.php

Most of this code is sort of duplicate of this one and does not require `exec()`.
It would require minor tweak to get rid of https://github.com/LycheeOrg/Lychee/blob/master/app/Metadata/GitHubFunctions.php#L72

**Comment by ildyria** [comment]
> I don't understand why `codecov/patch` is unhappy.

code patch wants at least 63% coverage on any new code.
I am actually considering nuking this check.

**Comment by nagmat84** [comment]
> https://github.com/LycheeOrg/Lychee/blob/master/app/Metadata/GitHubFunctions.php
> 
> Most of this code is sort of duplicate of this one and does not require `exec()`.
> It would require minor tweak to get rid of https://github.com/LycheeOrg/Lychee/blob/master/app/Metadata/GitHubFunctions.php#L72

Unfortunately, `GitHubFunctions::get_current_branch` claims to be on `master`, if unit tests are executed, even if the current branch is not `master`. So using that method does not help here, because it actually would not prevent the unlucky rebase.

IMHO, `GitHubFunctions::get_current_branch` should be modified to actually report the correct branch and not botch with the returned value, but I cannot foresee the implication of what would happen, if I changed that.

File: app/Metadata/GitHubFunctions.php [hunk 1]
```diff
@@ -69,10 +59,6 @@ private function trim($commit_id)
 	 */
 	public function get_current_branch()
 	{
-		if (App::runningUnitTests()) {
```
Comment (nagmat84) [inline app/Metadata/GitHubFunctions.php:72]: Are we sure that omitting that condition has no negative impact?

[`GitHubFunctions::count_behind(bool)`](https://github.com/LycheeOrg/Lychee/blob/bc918a394a995a939b6e5651746895c464370a2d/app/Metadata/GitHubFunctions.php#L127) checks for the master branch. `count_behind` is called in `is_up_to_date` and I have not checked further where this method is used. Anyway, I can't oversee right now (and haven't check) which other methods further down the call graph rely on `get_current_branch` to fake the master branch during tests.

File: app/Metadata/GitHubFunctions.php [hunk 2]
```diff
@@ -69,10 +59,6 @@ private function trim($commit_id)
 	 */
 	public function get_current_branch()
```
Comment (nagmat84) [inline app/Metadata/GitHubFunctions.php:59]: The PHPdoc comment needs updating. The last sentence

> Return master if it is CI

does not hold anymore.

**Review by nagmat84 — COMMENTED** [review]
In general, it looks good to me. My only hesitation comes from leaving out that one if-branch (see review comment in the affected source code). I would feel more confident about that, if someone could investigate and confirm that removing that if-branch has no negative impact.

File: app/Metadata/GitHubFunctions.php (hunk 1 above)
Comment (ildyria) [reply app/Metadata/GitHubFunctions.php:72]: Nah I don't think this impact in the count_behind. I would expect the tests to break otherwise. :)

Comment (nagmat84) [reply app/Metadata/GitHubFunctions.php:72]: OK, then let's be brave. If any bug occurs, we will fix it then.

File: app/Metadata/GitHubFunctions.php (hunk 2 above)
Comment (nagmat84) [reply app/Metadata/GitHubFunctions.php:59]: ok, it is updated.

**Pull request merged**